libs/dal/src/shared/services/cache.service.ts

```diff
@@ -0,0 +1,99 @@
+import Redis from 'ioredis';
+
+const STORE_CONNECTED = 'ready';
+
+export interface ICacheService {
+  set(key: string, value: string, options?: CachingConfig);
+  get(key: string);
+  del(key: string);
+  delByPattern(pattern: string);
+  keys(pattern?: string);
+  getStatus();
+  cacheEnabled();
+}
+
+export type CachingConfig = {
+  ttl?: number;
+};
+
+export class CacheService implements ICacheService {
+  private readonly client: Redis;
+  private readonly DEFAULT_TTL_SECONDS = 60;
+  private readonly cacheTtl: number;
+
+  constructor(
+    private config: {
+      cachePort: string;
+      cacheHost: string;
+      cachePassword?: string;
+      cacheConnectTimeout?: string;
+      cacheKeepAlive?: string;
+      cacheFamily?: string;
+      cacheKeyPrefix?: string;
+      cacheTtl?: string;
+    }
+  ) {
+    if (this.config.cachePort && this.config.cacheHost) {
+      this.client = new Redis(Number(this.config.cachePort), this.config.cacheHost, {
+        password: this.config.cachePassword,
+        connectTimeout: this.config.cacheConnectTimeout ? Number(this.config.cacheConnectTimeout) : 50000,
+        keepAlive: this.config.cacheKeepAlive ? Number(this.config.cacheKeepAlive) : 30000,
```

Comment:
Should we make the timeout and keep alive values potentially configurable storing them for now in constants?

Reply:
They are potentially and not mandatory, agree regarding the constants and I will add them shortly.

Reply:
I wrote such nonsense. What I meant is that now we can store them in constants and in the future we could make them configurable turning them into env variables so users could tweak values as needed.

Reply:
Oh I see well in any case we have them potentially configurable if needed. 🍻